### PR TITLE
fix:  Account deletion redirects too quickly

### LINKED ISSUE #39508: Account deletion redirects too quickly
**Describe the bug**
Deleting your account works as expected, but it also redirects users to /signin immediately after signing them out. This means the "Your account has successfully been deleted" message only appears for a split second before it disappears. I knew to look for it, but most users would not. Meaning they might misunderstand and think their account has not been deleted successfully.

**To Reproduce**
1. Go to the bottom of /settings
2. Click the "delete my account" button and confirm.
3. You will be redirected twice – once to the unauthenticated landing page, and then immediately afterward to /signin

### PR DESCRIPTION
**Pull request by souravv-singhh**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #39508

Added Time Delay function which will display account deletion message upto 3 seconds before redirecting.

### CODE REVIEW

File: client/src/redux/settings/danger-zone-saga.js [hunk 1]
```diff
@@ -7,15 +7,26 @@ import { postResetProgress, postDeleteAccount } from '../../utils/ajax';
...
+}
+
 function* deleteAccountSaga() {
   try {
     yield call(postDeleteAccount);
     yield put(
       createFlashMessage({
         type: 'info',
-        message: 'flash.account-deleted'
+        message: 'Your account has been successfully deleted'
```
Comment (Sembauke) [inline client/src/redux/settings/danger-zone-saga.js:25]: This message does not need to be changed, the value is stored in a key. 
```suggestion
        message: 'flash.account-deleted'
```

File: client/src/redux/settings/danger-zone-saga.js [hunk 2]
```diff
@@ -30,9 +41,11 @@ function* resetProgressSaga() {
     yield put(
       createFlashMessage({
         type: 'info',
-        message: 'flash.progress-reset'
+        message: 'Your progress has been reset'
```
Comment (Sembauke) [inline client/src/redux/settings/danger-zone-saga.js:44]:
```suggestion
        message: 'flash.progress-reset'
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: client/src/redux/settings/danger-zone-saga.js (hunk 1 above)
Comment (souravv-singhh) [reply client/src/redux/settings/danger-zone-saga.js]: sure thing!

File: client/src/redux/settings/danger-zone-saga.js (hunk 2 above)
Comment (souravv-singhh) [reply client/src/redux/settings/danger-zone-saga.js]: sure thing!

**Review by raisedadead — CHANGES_REQUESTED** [review]
Hi @souravv-singhh 

Thanks for taking the time to create this PR, IMHO, We do not want a time delay but would like to prevent the redirection to the sign-in page, instead, the user should be taken to the learn homepage and the notification that their account was deleted should be preserved.

**Comment by Sembauke** [comment]
Hey @souravv-singhh,
If you want to revisit this issue, feel free to re-open.

**Pull request closed without merging**